### PR TITLE
Preparation for stream

### PR DESCRIPTION
**Pull request by lukasz-pyrzyk**
According to the #10

### CODE REVIEW

**Comment by lukasz-pyrzyk** [comment]
@YPermitin I've finished serialization part. Can you please review?

File: src/BinaryFormatter/BinaryConverter.cs [hunk 1]
```diff
@@ -5,124 +5,80 @@
...
-            [typeof(float)] = new FloatConverter(),
-            [typeof(double)] = new DoubleConverter(),
-            [typeof(bool)] = new BoolConverter(),
-            [typeof(decimal)] = new DecimalConverter(),
-            [typeof(string)] = new StringConverter(),
-            [typeof(DateTime)] = new DatetimeConverter(),
-            [typeof(byte[])] = new ByteArrayConverter(),
-            [typeof(IEnumerable)] = new IEnumerableConverter()
-        };
+        private static readonly ConvertersSelector _selector = new ConvertersSelector();
```
Comment (YPermitin) [inline src/BinaryFormatter/BinaryConverter.cs]: I did the same in my experimental local branch ;)

**Comment by YPermitin** [comment]
@lukasz-pyrzyk You make a great job! Everything is good.
Streams looks nice =)

**Comment by lukasz-pyrzyk** [comment]
Ups, I've selected master as a target branch. Going to fix it :)

**Comment by YPermitin** [comment]
@lukasz-pyrzyk My fault. Did not notice.
We work on Agil, immediately all commits in production.

**Pull request merged**